### PR TITLE
Feature sigup login oauth (jwt, user creation and base model)  #13

### PR DESCRIPTION
**Pull request by jassalss**
I added the user creation, jwt and base model functionality. Please review and share your feedback.

### CODE REVIEW

File: server/models/base.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class BaseModel(ABC):
+
+    """
+        This will save the data contained in the inheriting classes 'data' dictionary
+         into a MongoDB collection named after the class that we are saving.
+    """
+    @abstractmethod
+    def get_dic(self):
```
Comment (sina-jamshidi) [inline server/models/base.py]: can we call this `get_dict()`. Typically in Python we refer to dictionaries as dicts (as you can see below with `dict()`).

File: server/models/base.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        pass
+
+    def save(self, collection_name):
```
Comment (sina-jamshidi) [inline server/models/base.py:15]: can we get this to handle `collection.update()` as well if the object already exists in the db

File: server/models/meeting.py [hunk 3]
```diff
@@ -10,17 +11,12 @@ def __init__(self, user_id, duration):
...
-        result = meetings.insert_one(meeting_data)
+        self.data = dict()
+        self.data['user_id'] = user_id
+        self.data['duration'] = duration
 
         if config.is_dev_environment():
-            print(f'DEBUG: Added meeting {result.inserted_id}')
+            print(f'DEBUG: Created meeting for user: {self.data["user_id"]}, length: {self.data["duration"]} minutes')
+
+    def get_dic():
```
Comment (sina-jamshidi) [inline server/models/meeting.py]: change to `get_dict()`

File: server/models/user.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    def __init__(self, userObj):
+        self.first_name = userObj.given_name
+        self.last_name = userObj.family_name
+        self.email = userObj.email
+        self.time_zone = strftime("%z", gmtime())
+        self.availability = []
+        self.unique_url = ""
+
+    def get_dic(self):
```
Comment (sina-jamshidi) [inline server/models/user.py]: `get_dict(self)`

**Review by sina-jamshidi — APPROVED** [review]
Sorry to keep adding to this but let's also add a save_many() so that we can do `db.collection.insertMany()` as well.

File: server/models/base.py (hunk 1 above)
Comment (jassalss) [reply server/models/base.py]: Hi sina. Now base model inherits from dict. Same functionality different way.

File: server/models/base.py (hunk 2 above)
Comment (jassalss) [reply server/models/base.py:15]: I will work on update part in morning.

File: server/models/user.py (hunk 4 above)
Comment (jassalss) [reply server/models/user.py]: Ok thanks

**Review by sina-jamshidi — APPROVED** [review]
Looks good!

**Pull request merged**